scripts/plot_operator_breakdown.py

```diff
@@ -65,25 +94,70 @@
 
 df = df.fillna(0)
 
+# Obtain copy for normalized plots
+df_norm = df.copy()
+
 # Calculate share of total execution time (i.e., longer running benchmark items are weighted more)
-df.loc["Total"] = df.sum() / df.count()
+df_norm.loc["Total"] = df_norm.sum() / df_norm.count()
 
 # Normalize data from nanoseconds to percentage of total cost (calculated by dividing the cells value by the total of
 # the row it appears in)
-df.iloc[:, 0:] = df.iloc[:, 0:].apply(lambda x: x / x.sum(), axis=1)
+df_norm.iloc[:, 0:] = df_norm.iloc[:, 0:].apply(lambda x: x / x.sum(), axis=1)
 
 # Print the dataframe for easy access to the raw numbers
-print(df)
+print(df_norm)
 
 # Drop all operators that do not exceed 1% in any query
-df = df[df > 0.01].dropna(axis="columns", how="all")
+df_norm = df_norm[df_norm > 0.01].dropna(axis="columns", how="all")
+df_norm_queries = df_norm.loc[df_norm.index != "Total"]
+df_norm_total = df_norm.loc[["Total"]]
 
 # Setup colorscheme - using cubehelix, which provides a color mapping that gracefully degrades to grayscale
-colors = sns.cubehelix_palette(n_colors=len(df), rot=2, reverse=True, light=0.9, dark=0.1, hue=1)
+colors = sns.cubehelix_palette(n_colors=len(df_norm), rot=2, reverse=True, light=0.9, dark=0.1, hue=1)
 cmap = LinearSegmentedColormap.from_list("my_colormap", colors)
 
-# Plot it
-ax = df.plot.bar(stacked=True, figsize=(2 + len(df) / 4, 4), colormap=cmap)
+# Factor to adapt the sizes and ratios to large benchmarks with many queries (e.g., Join Order Benchmark).
+# The sqrt() somewhat accomodates that we don't create extremely wide charts but still adapt.
+query_count_apaption = int(round(math.sqrt(len(df_norm))))
```

Comment:
No idea what you are doing here.

Reply:
Changed the comment. Better?

Reply:
I think I already got lost at the first sentence. What sizes and ratios? Is your goal to make the figure wider if we have more queries? If yes, that should be said first. Also, assuming that I understood the purpose correctly, why don't we make this linearly dependent on the number of queries? A wide chart is better than an illegible one, isn't it?

Reply:
I started with a linear scaling but that was already "suboptimal" for DS, but for JOB it did not work at all (I think). Either you really have tiny bars for "small" workloads as TPC-H with looks a little bit stupid as there is sufficient space. Or you have "nice" TPC-H and you need to zoom into the JOB graph every time to see the results.
That sqrt() somewhat solved that for me and created okay'ish graphs for all three benchmarks.

Reply:
(I don't want to promise is looks good for an ERP system with thousands of queries though :) )

Reply:
> We want the figure to be slightly wider if more queries are printed. `added_figure_width` grows sublinearly, using sqrt just happens to work well for the benchmarks that we tested it with.